Bottom: f89d098ac082c736f6971087f1c0a3bc99290c55
Top:    c3b851fc19c2ebb311ce40b4bcf0fcf31667d578
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-10 20:43:05 +0000

Refresh of gtk-gl-avoid-use-of-near-as-a

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 2503107..bd9c2e7 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -3163,8 +3163,8 @@ ghid_unproject_to_z_plane (int ex, int ey, Coord pcb_z, double *pcb_x, double *p
   double aspect = (double)widget->allocation.width / (double)widget->allocation.height;
   double width = 2. * aspect;
   double height = 2.;
-  double near = 1.;
-  /* double far = 24.; */
+  double near_plane = 1.;
+  /* double far_plane = 24.; */
 
   /* This is nasty beyond words, but I'm lazy and translating directly
    * from some untested maths I derived which used this notation */
@@ -3239,13 +3239,13 @@ ghid_unproject_to_z_plane (int ex, int ey, Coord pcb_z, double *pcb_x, double *p
   fvz = (float)pcb_z;
 
   /* This isn't really X and Y? */
-  x = (C * fvz + D) * 2. / width  * near + vpx * (K * fvz + L);
-  y = (G * fvz + H) * 2. / height * near + vpy * (K * fvz + L);
+  x = (C * fvz + D) * 2. / width  * near_plane + vpx * (K * fvz + L);
+  y = (G * fvz + H) * 2. / height * near_plane + vpy * (K * fvz + L);
 
-  mat[0][0] = -vpx * I - A * 2 / width / near;
-  mat[0][1] = -vpx * J - B * 2 / width / near;
-  mat[1][0] = -vpy * I - E * 2 / height / near;
-  mat[1][1] = -vpy * J - F * 2 / height / near;
+  mat[0][0] = -vpx * I - A * 2 / width / near_plane;
+  mat[0][1] = -vpx * J - B * 2 / width / near_plane;
+  mat[1][0] = -vpy * I - E * 2 / height / near_plane;
+  mat[1][1] = -vpy * J - F * 2 / height / near_plane;
 
 //  if (fabs (determinant_2x2 (mat)) < 0.000000000001)
 //    printf ("Determinant is quite small\n");
@@ -3282,10 +3282,10 @@ ghid_unproject_to_z_plane (int ex, int ey, Coord pcb_z, double *pcb_x, double *p
     /* float ew = last_modelview_matrix[0][3] * fvx + last_modelview_matrix[1][3] * fvy + last_modelview_matrix[2][3] * fvz + last_modelview_matrix[3][3]; */
 
 #if 0
-    if (-ez < near)
-      printf ("ez is closer than the near clipping plane, ez = %f\n", ez);
-    if (-ez > far)
-      printf ("ez is further than the near clipping plane, ez = %f\n", ez);
+    if (-ez < near_plane)
+      printf ("ez is closer than the near_plane clipping plane, ez = %f\n", ez);
+    if (-ez > far_plane)
+      printf ("ez is further than the near_plane clipping plane, ez = %f\n", ez);
 #endif
     if (-ez < 0) {
       // printf ("EZ IS BEHIND THE CAMERA !! ez = %f\n", ez);
